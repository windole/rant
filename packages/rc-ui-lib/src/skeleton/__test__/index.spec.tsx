import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Skeleton } from '..';

describe('Skeleton', () => {
  let wrapper;
  afterEach(() => {
    wrapper.unmount();
    jest.restoreAllMocks();
  });

  it('should render with row width array correctly', async () => {
    wrapper = mount(<Skeleton row={4} rowWidth={['100%', 30, '5rem']} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render with row height array correctly', async () => {
    wrapper = mount(<Skeleton row={4} rowHeight={['10px', 30, '1rem']} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render default slot when loading is false', () => {
    wrapper = mount(
      <Skeleton loading={false}>
        <div>Content</div>
      </Skeleton>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should change avatar size when using avatar-size prop', () => {
    wrapper = mount(<Skeleton avatar avatarSize="20rem" />);

    const avatar = wrapper.find('.rc-skeleton__avatar');
    expect(avatar.getDOMNode().style.width).toMatchSnapshot('20rem');
    expect(avatar.getDOMNode().style.height).toMatchSnapshot('20ren');
  });

  it('should change avatar shape when using avatar-shape prop', () => {
    wrapper = mount(<Skeleton avatar avatarShape="square" />);
    expect(toJson(wrapper.find('.rc-skeleton__avatar'))).toMatchSnapshot();
  });

  it('should be round when using round prop', () => {
    wrapper = mount(<Skeleton title round avatar />);
    expect(wrapper.find('.rc-skeleton--round').exists()).toBeTruthy();
  });

  it('should allow to disable animation', async () => {
    wrapper = mount(<Skeleton row={1} />);

    expect(wrapper.find('.rc-skeleton--animate').exists()).toBeTruthy();

    await wrapper.setProps({ animate: false });
    expect(wrapper.find('.rc-skeleton--animate').exists()).toBeFalsy();
  });
});
