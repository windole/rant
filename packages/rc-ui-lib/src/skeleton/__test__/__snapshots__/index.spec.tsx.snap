// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton should change avatar shape when using avatar-shape prop 1`] = `
<div
  className="rc-skeleton__avatar rc-skeleton__avatar--square"
  style={Object {}}
/>
`;

exports[`Skeleton should change avatar size when using avatar-size prop: 20rem 1`] = `"20rem"`;

exports[`Skeleton should change avatar size when using avatar-size prop: 20ren 1`] = `"20rem"`;

exports[`Skeleton should render default slot when loading is false 1`] = `
<Skeleton
  animate={true}
  avatarShape="round"
  loading={false}
  round={true}
  row={3}
  rowWidth="100%"
>
  <div>
    Content
  </div>
</Skeleton>
`;

exports[`Skeleton should render with row height array correctly 1`] = `
<Skeleton
  animate={true}
  avatarShape="round"
  loading={true}
  round={true}
  row={4}
  rowHeight={
    Array [
      "10px",
      30,
      "1rem",
    ]
  }
  rowWidth="100%"
>
  <div
    className="rc-skeleton rc-skeleton--animate rc-skeleton--round"
  >
    <div
      className="rc-skeleton__content"
    >
      <div
        className="rc-skeleton__row"
        key="0"
        style={
          Object {
            "height": "10px",
            "width": "100%",
          }
        }
      />
      <div
        className="rc-skeleton__row"
        key="1"
        style={
          Object {
            "height": "30px",
            "width": "100%",
          }
        }
      />
      <div
        className="rc-skeleton__row"
        key="2"
        style={
          Object {
            "height": "1rem",
            "width": "100%",
          }
        }
      />
      <div
        className="rc-skeleton__row"
        key="3"
        style={
          Object {
            "height": undefined,
            "width": "60%",
          }
        }
      />
    </div>
  </div>
</Skeleton>
`;

exports[`Skeleton should render with row width array correctly 1`] = `
<Skeleton
  animate={true}
  avatarShape="round"
  loading={true}
  round={true}
  row={4}
  rowWidth={
    Array [
      "100%",
      30,
      "5rem",
    ]
  }
>
  <div
    className="rc-skeleton rc-skeleton--animate rc-skeleton--round"
  >
    <div
      className="rc-skeleton__content"
    >
      <div
        className="rc-skeleton__row"
        key="0"
        style={
          Object {
            "height": undefined,
            "width": "100%",
          }
        }
      />
      <div
        className="rc-skeleton__row"
        key="1"
        style={
          Object {
            "height": undefined,
            "width": "30px",
          }
        }
      />
      <div
        className="rc-skeleton__row"
        key="2"
        style={
          Object {
            "height": undefined,
            "width": "5rem",
          }
        }
      />
      <div
        className="rc-skeleton__row"
        key="3"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
    </div>
  </div>
</Skeleton>
`;
